mscore/preferences.cpp

```diff
@@ -38,6 +38,29 @@ void Preferences::init(bool storeInMemoryOnly)
             _settings = new QSettings();
             }
 
+#if defined(Q_OS_WIN)
+      // Use the correct default GUI font on Windows. Qt 5 and below use the deprecated function GetStockObject() with
+      // DEFAULT_GUI_FONT, which returns MS Shell Dlg 2 in 8 pt. MS Shell Dlg 2 is a virtual font that maps to Tahoma, which has
+      // not been the default Windows GUI font since 2006.
+      //
+      // The correct way to determine the default GUI font is to call SystemParametersInfoW() with SPI_GETNONCLIENTMETRICS and use
+      // the returned lfMessageFont structure. On all versions of Windows from Windows Vista through Windows 10, this typically
+      // returns Segoe UI in 9 pt.
+      //
+      // This problem is slated to be fixed in Qt 6. For details, see: https://bugreports.qt.io/browse/QTBUG-58610
+      //
+      // In the meantime, we can work around the problem by having Qt use the "QMessageBox" font instead, which is already being
+      // initialized the correct way.
+      QApplication::setFont(QApplication::font("QMessageBox"));
```

Comment:
How about moving that into the if block below? If it is set already, and to something different than "MS Shell Dlg2", why initialize it?

Reply:
I think you might be misunderstanding what that line of code does.

It takes advantage of the fact that the font is already initialized correctly for the `QMessageBox` class but not for the overall application. So it queries Qt for the font used for the `QMessageBox` class and then calls `setFont()` to tell Qt to also use it as the default GUI font for the whole app.

It's equivalent to the following line:

`QApplication::setFont(QApplication::font("QMessageBox"), nullptr);`

The `nullptr` argument tells the function to set the default application font. It can be omitted because it's an optional parameter that already defaults to `nullptr`.

That single line of code replaces the manual initialization that I was doing in the previous version of this PR. It always needs to be done under Windows, regardless of the state of the settings file.